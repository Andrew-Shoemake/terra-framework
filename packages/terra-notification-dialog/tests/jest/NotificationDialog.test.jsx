import React from 'react';
/* eslint-disable import/no-extraneous-dependencies */
import { mountWithIntl, shallowWithIntl } from 'terra-enzyme-intl';
import { KEY_ESCAPE } from 'keycode-js';
import NotificationDialog from '../../src/NotificationDialog';

describe('Notification Dialog', () => {
  const acceptOnClick = jest.fn();
  const rejectOnClick = jest.fn();

  it('shallow renders an alert notification-dialog', () => {
    const dialog = shallowWithIntl(
      <NotificationDialog
        variant="alert"
        isOpen
        title="Test"
        startMessage="This text is used to provide more details."
        acceptAction={{ text: 'accept', onClick: () => {} }}
        rejectAction={{ text: 'reject', onClick: () => {} }}
      />,
    );

    expect(dialog).toMatchSnapshot();
    const actions = dialog.find('.actions');
    expect(actions.children().length).toBe(2);
    expect(actions.children().at(0).props()).toHaveProperty('data-terra-notification-dialog-button', 'accept');
    expect(actions.children().at(1).props()).toHaveProperty('data-terra-notification-dialog-button', 'reject');
  });

  it('shallow renders an error notification-dialog', () => {
    const dialog = shallowWithIntl(
      <NotificationDialog
        variant="error"
        isOpen
        title="Test"
        startMessage="This text is used to provide more details."
        acceptAction={{ text: 'accept', onClick: () => {} }}
        rejectAction={{ text: 'reject', onClick: () => {} }}
      />,
    );

    expect(dialog).toMatchSnapshot();
  });

  it('shallow renders an warning notification-dialog', () => {
    const dialog = shallowWithIntl(
      <NotificationDialog
        variant="warning"
        isOpen
        title="Test"
        startMessage="This text is used to provide more details."
        acceptAction={{ text: 'accept', onClick: () => {} }}
        rejectAction={{ text: 'reject', onClick: () => {} }}
      />,
    );

    expect(dialog).toMatchSnapshot();
  });

  it('shallow renders an info notification-dialog', () => {
    const dialog = shallowWithIntl(
      <NotificationDialog
        variant="info"
        isOpen
        title="Test"
        startMessage="This text is used to provide more details."
        acceptAction={{ text: 'accept', onClick: () => {} }}
        rejectAction={{ text: 'reject', onClick: () => {} }}
      />,
    );

    expect(dialog).toMatchSnapshot();
  });

  it('shallow renders an custom notification-dialog', () => {
    const dialog = shallowWithIntl(
      <NotificationDialog
        variant="custom"
        isOpen
        title="Test"
        startMessage="This text is used to provide more details."
        acceptAction={{ text: 'accept', onClick: () => {} }}
        rejectAction={{ text: 'reject', onClick: () => {} }}
        custom={{ signalWord: 'Custom', iconClassName: 'custom-icon' }}
      />,
    );

    expect(dialog).toMatchSnapshot();
  });

  it('shallow renders dialog with startMessage, content, and endMessage', () => {
    const dialog = shallowWithIntl(
      <NotificationDialog
        variant="alert"
        isOpen
        title="Test"
        startMessage="This text is used to provide more details."
        content={<div>Middle content</div>}
        endContent="end content before buttons"
        acceptAction={{ text: 'accept', onClick: () => {} }}
        rejectAction={{ text: 'reject', onClick: () => {} }}
      />,
    );

    expect(dialog).toMatchSnapshot();
  });

  it('shallow renders dialog with reject button first', () => {
    const dialog = shallowWithIntl(
      <NotificationDialog
        variant="alert"
        isOpen
        title="Test"
        startMessage="This text is used to provide more details."
        buttonOrder="rejectFirst"
        acceptAction={{ text: 'accept', onClick: acceptOnClick }}
        rejectAction={{ text: 'reject', onClick: rejectOnClick }}
      />,
    );

    expect(dialog).toMatchSnapshot();
    const actions = dialog.find('.actions');
    expect(actions.children().length).toBe(2);
    expect(actions.children().at(0).props()).toHaveProperty('data-terra-notification-dialog-button', 'reject');
    expect(actions.children().at(1).props()).toHaveProperty('data-terra-notification-dialog-button', 'accept');
  });

  it('correctly applies the theme context className', () => {
    jest.spyOn(React, 'useContext')
      .mockReturnValue({
        className: 'orion-fusion-theme',
      });

    const dialog = shallowWithIntl(
      <NotificationDialog
        variant="alert"
        isOpen
        startMessage="This text is used to provide more details."
        rejectAction={{ text: 'reject', onClick: rejectOnClick }}
      />,
    );

    expect(dialog).toMatchSnapshot();
  });

  it('throws error if variant is missing', () => {
    expect(() => shallowWithIntl(
      <NotificationDialog
        isOpen
        acceptAction={{ text: 'accept', onClick: acceptOnClick }}
      />,
    )).toThrowErrorMatchingSnapshot();
  });

  it('throws error if variant is missing', () => {
    expect(() => shallowWithIntl(
      <NotificationDialog
        isOpen
        variant="error"
      />,
    )).toThrowErrorMatchingSnapshot();
  });
});

describe('Notification Dialog functions as expected', () => {
  const acceptOnClick = jest.fn();
  const rejectOnClick = jest.fn();
  let dialogExample;
  beforeAll(() => {
    const Example = () => {
      const [isOpen, setIsOpen] = React.useState(false);
      return (
        <div>
          <NotificationDialog
            variant="alert"
            isOpen={isOpen}
            title="Test"
            startMessage="This text is used to provide more details."
            acceptAction={{ text: 'accept', onClick: acceptOnClick.mockImplementation(() => { setIsOpen(false); }) }}
            rejectAction={{ text: 'reject', onClick: rejectOnClick.mockImplementation(() => { setIsOpen(false); }) }}
          />
          <button type="button" data-test-button onClick={() => setIsOpen(true)}>Open Dialog</button>
        </div>
      );
    };
    dialogExample = mountWithIntl(<Example />);
  });

  it('should open the dialog', () => {
    expect(dialogExample).toMatchSnapshot();
    expect(dialogExample.find('NotificationDialog').props()).toHaveProperty('isOpen', false);
    expect(dialogExample.find('button[data-test-button]').length).toBe(1);
    dialogExample.find('button[data-test-button]').simulate('click');
    expect(dialogExample.find('NotificationDialog').props()).toHaveProperty('isOpen', true);
    expect(dialogExample.find('Button[data-terra-notification-dialog-button]').length).toBe(2);
  });

  it('should close the dialog when clicking reject button', () => {
    dialogExample.find('Button[data-terra-notification-dialog-button="reject"]').simulate('click');
    expect(rejectOnClick).toHaveBeenCalled();
    expect(dialogExample.find('NotificationDialog').props()).toHaveProperty('isOpen', false);
  });

  it('should close the dialog when clicking accept button', () => {
    // re-open
    dialogExample.find('button[data-test-button]').simulate('click');
    expect(dialogExample.find('NotificationDialog').props()).toHaveProperty('isOpen', true);

    dialogExample.find('Button[data-terra-notification-dialog-button="accept"]').simulate('click');
    expect(acceptOnClick).toHaveBeenCalled();
    expect(dialogExample.find('NotificationDialog').props()).toHaveProperty('isOpen', false);
  });

  it('should not close the dialog when pressing escape key', () => {
    // re-open
    dialogExample.find('button[data-test-button]').simulate('click');
    expect(dialogExample.find('NotificationDialog').props()).toHaveProperty('isOpen', true);

    dialogExample.find('Button[data-terra-notification-dialog-button="accept"]').simulate('keydown', { keycode: KEY_ESCAPE });
    expect(dialogExample.find('NotificationDialog').props()).toHaveProperty('isOpen', true);
  });
});
