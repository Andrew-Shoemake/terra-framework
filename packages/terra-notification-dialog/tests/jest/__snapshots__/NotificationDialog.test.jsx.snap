// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification Dialog correctly applies the theme context className 1`] = `
<AbstractModal
  aria-describedby="notification-dialog-header"
  aria-labelledby="notification-dialog-header"
  ariaLabel="Notification Dialog"
  classNameModal="notification-dialog orion-fusion-theme"
  classNameOverlay={null}
  closeOnEsc={false}
  closeOnOutsideClick={false}
  data-terra-notification-dialog={true}
  isFullscreen={false}
  isOpen={true}
  onRequestClose={[Function]}
  role="alertdialog"
  rootSelector="#root"
  zIndex="9000"
>
  <FocusTrap
    _createFocusTrap={[Function]}
    active={true}
    focusTrapOptions={
      Object {
        "clickOutsideDeactivates": false,
        "escapeDeactivates": false,
        "returnFocusOnDeactivate": true,
      }
    }
    paused={false}
  >
    <div
      className="notification-dialog-inner-wrapper"
    >
      <div
        className="notification-dialog-container"
        tabIndex="0"
      >
        <div
          className="floating-header-background alert"
        />
        <div
          className="header"
        >
          <div
            className="header-content"
          >
            <NotificationDialogIcon
              variant="alert"
            />
            <div
              className="header-container"
            >
              <div
                className="signal-word"
                id="notification-dialog-signal-word"
              >
                <FormattedMessage
                  id="Terra.notification.dialog.hazard-high"
                  values={Object {}}
                />
              </div>
              <div
                className="title"
                id="notification-dialog-title"
              />
            </div>
          </div>
        </div>
        <div
          className="body"
        >
          <div
            className="message"
          >
            This text is used to provide more details.
          </div>
        </div>
        <div
          className="footer"
        >
          <div
            className="actions"
          >
            <Button
              data-terra-notification-dialog-button="reject"
              isBlock={false}
              isCompact={false}
              isDisabled={false}
              isIconOnly={false}
              isReversed={false}
              key="reject"
              onClick={[MockFunction]}
              text="reject"
              type="button"
              variant="neutral"
            />
          </div>
        </div>
      </div>
    </div>
  </FocusTrap>
</AbstractModal>
`;

exports[`Notification Dialog functions as expected should open the dialog 1`] = `
<Example
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <div>
    <NotificationDialog
      acceptAction={
        Object {
          "onClick": [MockFunction],
          "text": "accept",
        }
      }
      buttonOrder="acceptFirst"
      custom={Object {}}
      dialogTitle="Test"
      emphasizedAction="none"
      isOpen={false}
      rejectAction={
        Object {
          "onClick": [MockFunction],
          "text": "reject",
        }
      }
      startMessage="This text is used to provide more details."
      variant="alert"
    >
      <AbstractModal
        aria-describedby="notification-dialog-title"
        aria-labelledby="notification-dialog-header"
        ariaLabel="Notification Dialog"
        classNameModal="notification-dialog orion-fusion-theme"
        classNameOverlay={null}
        closeOnEsc={false}
        closeOnOutsideClick={false}
        data-terra-notification-dialog={true}
        isFullscreen={false}
        isOpen={false}
        onRequestClose={[Function]}
        role="alertdialog"
        rootSelector="#root"
        zIndex="9000"
      />
    </NotificationDialog>
    <button
      data-test-button={true}
      onClick={[Function]}
      type="button"
    >
      Open Dialog
    </button>
  </div>
</Example>
`;

exports[`Notification Dialog shallow renders an alert notification-dialog 1`] = `
<AbstractModal
  aria-describedby="notification-dialog-title"
  aria-labelledby="notification-dialog-header"
  ariaLabel="Notification Dialog"
  classNameModal="notification-dialog"
  classNameOverlay={null}
  closeOnEsc={false}
  closeOnOutsideClick={false}
  data-terra-notification-dialog={true}
  isFullscreen={false}
  isOpen={true}
  onRequestClose={[Function]}
  role="alertdialog"
  rootSelector="#root"
  zIndex="9000"
>
  <FocusTrap
    _createFocusTrap={[Function]}
    active={true}
    focusTrapOptions={
      Object {
        "clickOutsideDeactivates": false,
        "escapeDeactivates": false,
        "returnFocusOnDeactivate": true,
      }
    }
    paused={false}
  >
    <div
      className="notification-dialog-inner-wrapper"
    >
      <div
        className="notification-dialog-container"
        tabIndex="0"
      >
        <div
          className="floating-header-background alert"
        />
        <div
          className="header"
        >
          <div
            className="header-content"
          >
            <NotificationDialogIcon
              variant="alert"
            />
            <div
              className="header-container"
            >
              <div
                className="signal-word"
                id="notification-dialog-signal-word"
              >
                <FormattedMessage
                  id="Terra.notification.dialog.hazard-high"
                  values={Object {}}
                />
              </div>
              <div
                className="title"
                id="notification-dialog-title"
              >
                Test
              </div>
            </div>
          </div>
        </div>
        <div
          className="body"
        >
          <div
            className="message"
          >
            This text is used to provide more details.
          </div>
        </div>
        <div
          className="footer"
        >
          <div
            className="actions"
          >
            <Button
              data-terra-notification-dialog-button="accept"
              isBlock={false}
              isCompact={false}
              isDisabled={false}
              isIconOnly={false}
              isReversed={false}
              key="accept"
              onClick={[Function]}
              text="accept"
              type="button"
              variant="neutral"
            />
            <Button
              data-terra-notification-dialog-button="reject"
              isBlock={false}
              isCompact={false}
              isDisabled={false}
              isIconOnly={false}
              isReversed={false}
              key="reject"
              onClick={[Function]}
              text="reject"
              type="button"
              variant="neutral"
            />
          </div>
        </div>
      </div>
    </div>
  </FocusTrap>
</AbstractModal>
`;

exports[`Notification Dialog shallow renders an custom notification-dialog 1`] = `
<AbstractModal
  aria-describedby="notification-dialog-title"
  aria-labelledby="notification-dialog-header"
  ariaLabel="Notification Dialog"
  classNameModal="notification-dialog"
  classNameOverlay={null}
  closeOnEsc={false}
  closeOnOutsideClick={false}
  data-terra-notification-dialog={true}
  isFullscreen={false}
  isOpen={true}
  onRequestClose={[Function]}
  role="alertdialog"
  rootSelector="#root"
  zIndex="9000"
>
  <FocusTrap
    _createFocusTrap={[Function]}
    active={true}
    focusTrapOptions={
      Object {
        "clickOutsideDeactivates": false,
        "escapeDeactivates": false,
        "returnFocusOnDeactivate": true,
      }
    }
    paused={false}
  >
    <div
      className="notification-dialog-inner-wrapper"
    >
      <div
        className="notification-dialog-container"
        tabIndex="0"
      >
        <div
          className="floating-header-background custom"
        />
        <div
          className="header"
        >
          <div
            className="header-content"
          >
            <NotificationDialogIcon
              iconClassName="custom-icon"
              variant="custom"
            />
            <div
              className="header-container"
            >
              <div
                className="signal-word"
                id="notification-dialog-signal-word"
              >
                Custom
              </div>
              <div
                className="title"
                id="notification-dialog-title"
              >
                Test
              </div>
            </div>
          </div>
        </div>
        <div
          className="body"
        >
          <div
            className="message"
          >
            This text is used to provide more details.
          </div>
        </div>
        <div
          className="footer"
        >
          <div
            className="actions"
          >
            <Button
              data-terra-notification-dialog-button="accept"
              isBlock={false}
              isCompact={false}
              isDisabled={false}
              isIconOnly={false}
              isReversed={false}
              key="accept"
              onClick={[Function]}
              text="accept"
              type="button"
              variant="neutral"
            />
            <Button
              data-terra-notification-dialog-button="reject"
              isBlock={false}
              isCompact={false}
              isDisabled={false}
              isIconOnly={false}
              isReversed={false}
              key="reject"
              onClick={[Function]}
              text="reject"
              type="button"
              variant="neutral"
            />
          </div>
        </div>
      </div>
    </div>
  </FocusTrap>
</AbstractModal>
`;

exports[`Notification Dialog shallow renders an error notification-dialog 1`] = `
<AbstractModal
  aria-describedby="notification-dialog-title"
  aria-labelledby="notification-dialog-header"
  ariaLabel="Notification Dialog"
  classNameModal="notification-dialog"
  classNameOverlay={null}
  closeOnEsc={false}
  closeOnOutsideClick={false}
  data-terra-notification-dialog={true}
  isFullscreen={false}
  isOpen={true}
  onRequestClose={[Function]}
  role="alertdialog"
  rootSelector="#root"
  zIndex="9000"
>
  <FocusTrap
    _createFocusTrap={[Function]}
    active={true}
    focusTrapOptions={
      Object {
        "clickOutsideDeactivates": false,
        "escapeDeactivates": false,
        "returnFocusOnDeactivate": true,
      }
    }
    paused={false}
  >
    <div
      className="notification-dialog-inner-wrapper"
    >
      <div
        className="notification-dialog-container"
        tabIndex="0"
      >
        <div
          className="floating-header-background error"
        />
        <div
          className="header"
        >
          <div
            className="header-content"
          >
            <NotificationDialogIcon
              variant="error"
            />
            <div
              className="header-container"
            >
              <div
                className="signal-word"
                id="notification-dialog-signal-word"
              >
                <FormattedMessage
                  id="Terra.notification.dialog.error"
                  values={Object {}}
                />
              </div>
              <div
                className="title"
                id="notification-dialog-title"
              >
                Test
              </div>
            </div>
          </div>
        </div>
        <div
          className="body"
        >
          <div
            className="message"
          >
            This text is used to provide more details.
          </div>
        </div>
        <div
          className="footer"
        >
          <div
            className="actions"
          >
            <Button
              data-terra-notification-dialog-button="accept"
              isBlock={false}
              isCompact={false}
              isDisabled={false}
              isIconOnly={false}
              isReversed={false}
              key="accept"
              onClick={[Function]}
              text="accept"
              type="button"
              variant="neutral"
            />
            <Button
              data-terra-notification-dialog-button="reject"
              isBlock={false}
              isCompact={false}
              isDisabled={false}
              isIconOnly={false}
              isReversed={false}
              key="reject"
              onClick={[Function]}
              text="reject"
              type="button"
              variant="neutral"
            />
          </div>
        </div>
      </div>
    </div>
  </FocusTrap>
</AbstractModal>
`;

exports[`Notification Dialog shallow renders an info notification-dialog 1`] = `
<AbstractModal
  aria-describedby="notification-dialog-title"
  aria-labelledby="notification-dialog-header"
  ariaLabel="Notification Dialog"
  classNameModal="notification-dialog"
  classNameOverlay={null}
  closeOnEsc={false}
  closeOnOutsideClick={false}
  data-terra-notification-dialog={true}
  isFullscreen={false}
  isOpen={true}
  onRequestClose={[Function]}
  role="alertdialog"
  rootSelector="#root"
  zIndex="9000"
>
  <FocusTrap
    _createFocusTrap={[Function]}
    active={true}
    focusTrapOptions={
      Object {
        "clickOutsideDeactivates": false,
        "escapeDeactivates": false,
        "returnFocusOnDeactivate": true,
      }
    }
    paused={false}
  >
    <div
      className="notification-dialog-inner-wrapper"
    >
      <div
        className="notification-dialog-container"
        tabIndex="0"
      >
        <div
          className="floating-header-background info"
        />
        <div
          className="header"
        >
          <div
            className="header-content"
          >
            <NotificationDialogIcon
              variant="info"
            />
            <div
              className="header-container"
            >
              <div
                className="signal-word"
                id="notification-dialog-signal-word"
              >
                <FormattedMessage
                  id="Terra.notification.dialog.hazard-low"
                  values={Object {}}
                />
              </div>
              <div
                className="title"
                id="notification-dialog-title"
              >
                Test
              </div>
            </div>
          </div>
        </div>
        <div
          className="body"
        >
          <div
            className="message"
          >
            This text is used to provide more details.
          </div>
        </div>
        <div
          className="footer"
        >
          <div
            className="actions"
          >
            <Button
              data-terra-notification-dialog-button="accept"
              isBlock={false}
              isCompact={false}
              isDisabled={false}
              isIconOnly={false}
              isReversed={false}
              key="accept"
              onClick={[Function]}
              text="accept"
              type="button"
              variant="neutral"
            />
            <Button
              data-terra-notification-dialog-button="reject"
              isBlock={false}
              isCompact={false}
              isDisabled={false}
              isIconOnly={false}
              isReversed={false}
              key="reject"
              onClick={[Function]}
              text="reject"
              type="button"
              variant="neutral"
            />
          </div>
        </div>
      </div>
    </div>
  </FocusTrap>
</AbstractModal>
`;

exports[`Notification Dialog shallow renders an warning notification-dialog 1`] = `
<AbstractModal
  aria-describedby="notification-dialog-title"
  aria-labelledby="notification-dialog-header"
  ariaLabel="Notification Dialog"
  classNameModal="notification-dialog"
  classNameOverlay={null}
  closeOnEsc={false}
  closeOnOutsideClick={false}
  data-terra-notification-dialog={true}
  isFullscreen={false}
  isOpen={true}
  onRequestClose={[Function]}
  role="alertdialog"
  rootSelector="#root"
  zIndex="9000"
>
  <FocusTrap
    _createFocusTrap={[Function]}
    active={true}
    focusTrapOptions={
      Object {
        "clickOutsideDeactivates": false,
        "escapeDeactivates": false,
        "returnFocusOnDeactivate": true,
      }
    }
    paused={false}
  >
    <div
      className="notification-dialog-inner-wrapper"
    >
      <div
        className="notification-dialog-container"
        tabIndex="0"
      >
        <div
          className="floating-header-background warning"
        />
        <div
          className="header"
        >
          <div
            className="header-content"
          >
            <NotificationDialogIcon
              variant="warning"
            />
            <div
              className="header-container"
            >
              <div
                className="signal-word"
                id="notification-dialog-signal-word"
              >
                <FormattedMessage
                  id="Terra.notification.dialog.hazard-medium"
                  values={Object {}}
                />
              </div>
              <div
                className="title"
                id="notification-dialog-title"
              >
                Test
              </div>
            </div>
          </div>
        </div>
        <div
          className="body"
        >
          <div
            className="message"
          >
            This text is used to provide more details.
          </div>
        </div>
        <div
          className="footer"
        >
          <div
            className="actions"
          >
            <Button
              data-terra-notification-dialog-button="accept"
              isBlock={false}
              isCompact={false}
              isDisabled={false}
              isIconOnly={false}
              isReversed={false}
              key="accept"
              onClick={[Function]}
              text="accept"
              type="button"
              variant="neutral"
            />
            <Button
              data-terra-notification-dialog-button="reject"
              isBlock={false}
              isCompact={false}
              isDisabled={false}
              isIconOnly={false}
              isReversed={false}
              key="reject"
              onClick={[Function]}
              text="reject"
              type="button"
              variant="neutral"
            />
          </div>
        </div>
      </div>
    </div>
  </FocusTrap>
</AbstractModal>
`;

exports[`Notification Dialog shallow renders dialog with reject button first 1`] = `
<AbstractModal
  aria-describedby="notification-dialog-title"
  aria-labelledby="notification-dialog-header"
  ariaLabel="Notification Dialog"
  classNameModal="notification-dialog"
  classNameOverlay={null}
  closeOnEsc={false}
  closeOnOutsideClick={false}
  data-terra-notification-dialog={true}
  isFullscreen={false}
  isOpen={true}
  onRequestClose={[Function]}
  role="alertdialog"
  rootSelector="#root"
  zIndex="9000"
>
  <FocusTrap
    _createFocusTrap={[Function]}
    active={true}
    focusTrapOptions={
      Object {
        "clickOutsideDeactivates": false,
        "escapeDeactivates": false,
        "returnFocusOnDeactivate": true,
      }
    }
    paused={false}
  >
    <div
      className="notification-dialog-inner-wrapper"
    >
      <div
        className="notification-dialog-container"
        tabIndex="0"
      >
        <div
          className="floating-header-background alert"
        />
        <div
          className="header"
        >
          <div
            className="header-content"
          >
            <NotificationDialogIcon
              variant="alert"
            />
            <div
              className="header-container"
            >
              <div
                className="signal-word"
                id="notification-dialog-signal-word"
              >
                <FormattedMessage
                  id="Terra.notification.dialog.hazard-high"
                  values={Object {}}
                />
              </div>
              <div
                className="title"
                id="notification-dialog-title"
              >
                Test
              </div>
            </div>
          </div>
        </div>
        <div
          className="body"
        >
          <div
            className="message"
          >
            This text is used to provide more details.
          </div>
        </div>
        <div
          className="footer"
        >
          <div
            className="actions"
          >
            <Button
              data-terra-notification-dialog-button="reject"
              isBlock={false}
              isCompact={false}
              isDisabled={false}
              isIconOnly={false}
              isReversed={false}
              key="reject"
              onClick={[MockFunction]}
              text="reject"
              type="button"
              variant="neutral"
            />
            <Button
              data-terra-notification-dialog-button="accept"
              isBlock={false}
              isCompact={false}
              isDisabled={false}
              isIconOnly={false}
              isReversed={false}
              key="accept"
              onClick={[MockFunction]}
              text="accept"
              type="button"
              variant="neutral"
            />
          </div>
        </div>
      </div>
    </div>
  </FocusTrap>
</AbstractModal>
`;

exports[`Notification Dialog shallow renders dialog with startMessage, content, and endMessage 1`] = `
<AbstractModal
  aria-describedby="notification-dialog-title"
  aria-labelledby="notification-dialog-header"
  ariaLabel="Notification Dialog"
  classNameModal="notification-dialog"
  classNameOverlay={null}
  closeOnEsc={false}
  closeOnOutsideClick={false}
  data-terra-notification-dialog={true}
  isFullscreen={false}
  isOpen={true}
  onRequestClose={[Function]}
  role="alertdialog"
  rootSelector="#root"
  zIndex="9000"
>
  <FocusTrap
    _createFocusTrap={[Function]}
    active={true}
    focusTrapOptions={
      Object {
        "clickOutsideDeactivates": false,
        "escapeDeactivates": false,
        "returnFocusOnDeactivate": true,
      }
    }
    paused={false}
  >
    <div
      className="notification-dialog-inner-wrapper"
    >
      <div
        className="notification-dialog-container"
        tabIndex="0"
      >
        <div
          className="floating-header-background alert"
        />
        <div
          className="header"
        >
          <div
            className="header-content"
          >
            <NotificationDialogIcon
              variant="alert"
            />
            <div
              className="header-container"
            >
              <div
                className="signal-word"
                id="notification-dialog-signal-word"
              >
                <FormattedMessage
                  id="Terra.notification.dialog.hazard-high"
                  values={Object {}}
                />
              </div>
              <div
                className="title"
                id="notification-dialog-title"
              >
                Test
              </div>
            </div>
          </div>
        </div>
        <div
          className="body"
        >
          <div
            className="message"
          >
            This text is used to provide more details.
          </div>
          <div
            className="message"
          >
            <div>
              Middle content
            </div>
          </div>
        </div>
        <div
          className="footer"
        >
          <div
            className="actions"
          >
            <Button
              data-terra-notification-dialog-button="accept"
              isBlock={false}
              isCompact={false}
              isDisabled={false}
              isIconOnly={false}
              isReversed={false}
              key="accept"
              onClick={[Function]}
              text="accept"
              type="button"
              variant="neutral"
            />
            <Button
              data-terra-notification-dialog-button="reject"
              isBlock={false}
              isCompact={false}
              isDisabled={false}
              isIconOnly={false}
              isReversed={false}
              key="reject"
              onClick={[Function]}
              text="reject"
              type="button"
              variant="neutral"
            />
          </div>
        </div>
      </div>
    </div>
  </FocusTrap>
</AbstractModal>
`;
